src/diffusers/loaders/lora.py

```diff
@@ -959,7 +960,7 @@ def set_adapters_for_text_encoder(
         self,
         adapter_names: Union[List[str], str],
         text_encoder: Optional["PreTrainedModel"] = None,  # noqa: F821
-        text_encoder_weights: List[float] = None,
+        text_encoder_weights: Optional[Union[float, List[float], List[NoneType]]] = None,
```

Comment:
What does `List[NoneType]` signify? What kind of inputs are expected there?

Reply:
This function can receive a list of `None`s. Then for each adapter the default value of `1.0` is taken.

I've changes this because `LoraLoaderMixin.set_adapters_for_text_encoder` calls this function and can now pass a list of Nones.

We need to either add the type hint `List[NoneType]` here or add
```
if all(w is None for w in text_encoder_weights):
    text_encoder_weights = None
```
in `LoraLoaderMixin.set_adapters_for_text_encoder`

Reply:
Thanks for explaining!